{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0553d129-d122-4e38-b07d-259ce3938b37",
   "metadata": {},
   "source": [
    "## Map PGS ID to general disease category\n",
    "This script loads metadata from PGS catalog and from EFO database, and maps PGS to more general disease categories (e.g., melanoma -> cancer, obesity -> metabolic disorder).\n",
    "\n",
    "The first step after loading the data is to create a dictionary structure (key -> value) mapping PGS as keys and the most general EFO category names as values.\n",
    "\n",
    "After that, the script loads your typical PGSwas or PRSwas results (p-values or betas), parses the PGS ids from the PRS_name column, and creates a new column in your data called \"Trait_category\" with the more general disease category names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "31e2f56d-ee90-40e2-93b1-c927938b9aba",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fcaccbfd-569a-4164-bafb-6a52c7f7ac4e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#these are mapping tables that we will need to map the genetic risk ids (PGS) into the trait categories or experimental factor ontology (EFO) category.\n",
    "pgs_metadata_df = pd.read_csv(\"/blue/raquel.dias/raquel.dias/UKB/PGS_catalog/pgs_all_metadata_scores.csv\")\n",
    "trait_category_df = pd.read_csv(\"/blue/raquel.dias/raquel.dias/UKB/PGS_catalog/pgs_traits_data.csv\")\n",
    "\n",
    "#clean additional text following the EFO ids\n",
    "trait_category_df['Trait Identifier (ontology ID)'] = trait_category_df['Trait Identifier (ontology ID)'].str.replace(r':.*', '', regex=True)\n",
    "\n",
    "#cleaning traits that match to multiple categories, keeping first category only\n",
    "trait_category_df['Trait Category'] = trait_category_df['Trait Category'].str.replace(r\"(\\w)([A-Z]).*\", r\"\\1\", regex=True)\n",
    "\n",
    "#clean PGS ids that match to multiple EFO ids, keeping first EFO id only\n",
    "pgs_metadata_df['Mapped Trait(s) (EFO ID)'] = pgs_metadata_df['Mapped Trait(s) (EFO ID)'].str.replace(r'\\|.*', '', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e5e1b07e-c6b9-44b0-bf8d-bae2064a9d29",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Polygenic Score (PGS) ID</th>\n",
       "      <th>PGS Name</th>\n",
       "      <th>Reported Trait</th>\n",
       "      <th>Mapped Trait(s) (EFO label)</th>\n",
       "      <th>Mapped Trait(s) (EFO ID)</th>\n",
       "      <th>PGS Development Method</th>\n",
       "      <th>PGS Development Details/Relevant Parameters</th>\n",
       "      <th>Original Genome Build</th>\n",
       "      <th>Number of Variants</th>\n",
       "      <th>Number of Interaction Terms</th>\n",
       "      <th>...</th>\n",
       "      <th>PGS Publication (PGP) ID</th>\n",
       "      <th>Publication (PMID)</th>\n",
       "      <th>Publication (doi)</th>\n",
       "      <th>Score and results match the original publication</th>\n",
       "      <th>Ancestry Distribution (%) - Source of Variant Associations (GWAS)</th>\n",
       "      <th>Ancestry Distribution (%) - Score Development/Training</th>\n",
       "      <th>Ancestry Distribution (%) - PGS Evaluation</th>\n",
       "      <th>FTP link</th>\n",
       "      <th>Release Date</th>\n",
       "      <th>License/Terms of Use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PGS000001</td>\n",
       "      <td>PRS77_BC</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>EFO_0000305</td>\n",
       "      <td>SNPs passing genome-wide significance</td>\n",
       "      <td>P&lt;5x10-8</td>\n",
       "      <td>NR</td>\n",
       "      <td>77</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000001</td>\n",
       "      <td>25855707.0</td>\n",
       "      <td>10.1093/jnci/djv036</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>European:100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2019-10-14</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PGS000002</td>\n",
       "      <td>PRS77_ERpos</td>\n",
       "      <td>ER-positive Breast Cancer</td>\n",
       "      <td>estrogen-receptor positive breast cancer</td>\n",
       "      <td>EFO_1000649</td>\n",
       "      <td>SNPs passing genome-wide significance</td>\n",
       "      <td>P&lt;5x10-8</td>\n",
       "      <td>NR</td>\n",
       "      <td>77</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000001</td>\n",
       "      <td>25855707.0</td>\n",
       "      <td>10.1093/jnci/djv036</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>European:100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2019-10-14</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PGS000003</td>\n",
       "      <td>PRS77_ERneg</td>\n",
       "      <td>ER-negative Breast Cancer</td>\n",
       "      <td>estrogen-receptor negative breast cancer</td>\n",
       "      <td>EFO_1000650</td>\n",
       "      <td>SNPs passing genome-wide significance</td>\n",
       "      <td>P&lt;5x10-8</td>\n",
       "      <td>NR</td>\n",
       "      <td>77</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000001</td>\n",
       "      <td>25855707.0</td>\n",
       "      <td>10.1093/jnci/djv036</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>European:100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2019-10-14</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PGS000004</td>\n",
       "      <td>PRS313_BC</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>breast carcinoma</td>\n",
       "      <td>EFO_0000305</td>\n",
       "      <td>Hard-Thresholding Stepwise Forward Regression</td>\n",
       "      <td>p &lt; 10^-5</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>313</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000002</td>\n",
       "      <td>30554720.0</td>\n",
       "      <td>10.1016/j.ajhg.2018.11.002</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>European:100</td>\n",
       "      <td>European:76.1|Multi-ancestry (including Europe...</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2019-10-14</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PGS000005</td>\n",
       "      <td>PRS313_ERpos</td>\n",
       "      <td>ER-positive Breast Cancer</td>\n",
       "      <td>estrogen-receptor positive breast cancer</td>\n",
       "      <td>EFO_1000649</td>\n",
       "      <td>Hard-Thresholding Stepwise Forward Regression</td>\n",
       "      <td>p &lt; 10^-5</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>313</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000002</td>\n",
       "      <td>30554720.0</td>\n",
       "      <td>10.1016/j.ajhg.2018.11.002</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>European:100</td>\n",
       "      <td>European:90|Not Reported:5|Multi-ancestry (exc...</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2019-10-14</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3802</th>\n",
       "      <td>PGS003967</td>\n",
       "      <td>SBP_EUR</td>\n",
       "      <td>Systolic blood pressure</td>\n",
       "      <td>systolic blood pressure</td>\n",
       "      <td>EFO_0006335</td>\n",
       "      <td>PRSice2, LDPred2-Auto</td>\n",
       "      <td>R2 = 0.1</td>\n",
       "      <td>NR</td>\n",
       "      <td>1108368</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000510</td>\n",
       "      <td>37268629.0</td>\n",
       "      <td>10.1038/s41467-023-38990-9</td>\n",
       "      <td>True</td>\n",
       "      <td>European:81|East Asian:11.3|African:5.2|Hispan...</td>\n",
       "      <td>European:49.5|African:23.6|Hispanic or Latin A...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2023-10-17</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>PGS003968</td>\n",
       "      <td>SBP_weightedPRSsum</td>\n",
       "      <td>Systolic blood pressure</td>\n",
       "      <td>systolic blood pressure</td>\n",
       "      <td>EFO_0006335</td>\n",
       "      <td>PRSice2, LDPred2-Auto, PRS-CSx</td>\n",
       "      <td>PRSsum = UKBB+ICBP (PRS1), MVP, (PRS2), BBJ (P...</td>\n",
       "      <td>NR</td>\n",
       "      <td>1267240</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000510</td>\n",
       "      <td>37268629.0</td>\n",
       "      <td>10.1038/s41467-023-38990-9</td>\n",
       "      <td>True</td>\n",
       "      <td>European:79|East Asian:11|African:7.6|Hispanic...</td>\n",
       "      <td>European:62.6|African:16.8|Hispanic or Latin A...</td>\n",
       "      <td>Multi-ancestry (including European):100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2023-10-17</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>PGS003969</td>\n",
       "      <td>PRS39_HF</td>\n",
       "      <td>Heart failure</td>\n",
       "      <td>heart failure</td>\n",
       "      <td>EFO_0003144</td>\n",
       "      <td>Genome-wide significant SNPs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000511</td>\n",
       "      <td>37429843.0</td>\n",
       "      <td>10.1038/s41467-023-39253-3</td>\n",
       "      <td>True</td>\n",
       "      <td>European:100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>European:100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2023-10-17</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3805</th>\n",
       "      <td>PGS003970</td>\n",
       "      <td>HARE_META_SBP_PRS_MALE</td>\n",
       "      <td>Systolic Blood Pressure</td>\n",
       "      <td>systolic blood pressure</td>\n",
       "      <td>EFO_0006335</td>\n",
       "      <td>PRS-CS</td>\n",
       "      <td>Hapmap only variants with EUR as LD reference</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>1115602</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000512</td>\n",
       "      <td>37807951.0</td>\n",
       "      <td>10.1161/circgen.123.004259</td>\n",
       "      <td>True</td>\n",
       "      <td>European:94.1|Multi-ancestry (excluding Europe...</td>\n",
       "      <td>Multi-ancestry (including European):100</td>\n",
       "      <td>Multi-ancestry (including European):100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2023-10-17</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3806</th>\n",
       "      <td>PGS003971</td>\n",
       "      <td>HARE_META_SBP_PRS_FEMALE</td>\n",
       "      <td>Systolic Blood Pressure</td>\n",
       "      <td>systolic blood pressure</td>\n",
       "      <td>EFO_0006335</td>\n",
       "      <td>PRS-CS</td>\n",
       "      <td>Hapmap only variants with EUR as LD reference</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>1115520</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>PGP000512</td>\n",
       "      <td>37807951.0</td>\n",
       "      <td>10.1161/circgen.123.004259</td>\n",
       "      <td>True</td>\n",
       "      <td>European:94.4|Multi-ancestry (excluding Europe...</td>\n",
       "      <td>Multi-ancestry (including European):100</td>\n",
       "      <td>Multi-ancestry (including European):100</td>\n",
       "      <td>https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...</td>\n",
       "      <td>2023-10-17</td>\n",
       "      <td>PGS obtained from the Catalog should be cited ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Polygenic Score (PGS) ID                  PGS Name  \\\n",
       "0                   PGS000001                  PRS77_BC   \n",
       "1                   PGS000002               PRS77_ERpos   \n",
       "2                   PGS000003               PRS77_ERneg   \n",
       "3                   PGS000004                 PRS313_BC   \n",
       "4                   PGS000005              PRS313_ERpos   \n",
       "...                       ...                       ...   \n",
       "3802                PGS003967                   SBP_EUR   \n",
       "3803                PGS003968        SBP_weightedPRSsum   \n",
       "3804                PGS003969                  PRS39_HF   \n",
       "3805                PGS003970    HARE_META_SBP_PRS_MALE   \n",
       "3806                PGS003971  HARE_META_SBP_PRS_FEMALE   \n",
       "\n",
       "                 Reported Trait               Mapped Trait(s) (EFO label)  \\\n",
       "0                 Breast Cancer                          breast carcinoma   \n",
       "1     ER-positive Breast Cancer  estrogen-receptor positive breast cancer   \n",
       "2     ER-negative Breast Cancer  estrogen-receptor negative breast cancer   \n",
       "3                 Breast Cancer                          breast carcinoma   \n",
       "4     ER-positive Breast Cancer  estrogen-receptor positive breast cancer   \n",
       "...                         ...                                       ...   \n",
       "3802    Systolic blood pressure                   systolic blood pressure   \n",
       "3803    Systolic blood pressure                   systolic blood pressure   \n",
       "3804              Heart failure                             heart failure   \n",
       "3805    Systolic Blood Pressure                   systolic blood pressure   \n",
       "3806    Systolic Blood Pressure                   systolic blood pressure   \n",
       "\n",
       "     Mapped Trait(s) (EFO ID)                         PGS Development Method  \\\n",
       "0                 EFO_0000305          SNPs passing genome-wide significance   \n",
       "1                 EFO_1000649          SNPs passing genome-wide significance   \n",
       "2                 EFO_1000650          SNPs passing genome-wide significance   \n",
       "3                 EFO_0000305  Hard-Thresholding Stepwise Forward Regression   \n",
       "4                 EFO_1000649  Hard-Thresholding Stepwise Forward Regression   \n",
       "...                       ...                                            ...   \n",
       "3802              EFO_0006335                          PRSice2, LDPred2-Auto   \n",
       "3803              EFO_0006335                 PRSice2, LDPred2-Auto, PRS-CSx   \n",
       "3804              EFO_0003144                   Genome-wide significant SNPs   \n",
       "3805              EFO_0006335                                         PRS-CS   \n",
       "3806              EFO_0006335                                         PRS-CS   \n",
       "\n",
       "            PGS Development Details/Relevant Parameters Original Genome Build  \\\n",
       "0                                              P<5x10-8                    NR   \n",
       "1                                              P<5x10-8                    NR   \n",
       "2                                              P<5x10-8                    NR   \n",
       "3                                             p < 10^-5                GRCh37   \n",
       "4                                             p < 10^-5                GRCh37   \n",
       "...                                                 ...                   ...   \n",
       "3802                                           R2 = 0.1                    NR   \n",
       "3803  PRSsum = UKBB+ICBP (PRS1), MVP, (PRS2), BBJ (P...                    NR   \n",
       "3804                                                NaN                    NR   \n",
       "3805      Hapmap only variants with EUR as LD reference                GRCh38   \n",
       "3806      Hapmap only variants with EUR as LD reference                GRCh38   \n",
       "\n",
       "      Number of Variants  Number of Interaction Terms  ...  \\\n",
       "0                     77                            0  ...   \n",
       "1                     77                            0  ...   \n",
       "2                     77                            0  ...   \n",
       "3                    313                            0  ...   \n",
       "4                    313                            0  ...   \n",
       "...                  ...                          ...  ...   \n",
       "3802             1108368                            0  ...   \n",
       "3803             1267240                            0  ...   \n",
       "3804                  39                            0  ...   \n",
       "3805             1115602                            0  ...   \n",
       "3806             1115520                            0  ...   \n",
       "\n",
       "     PGS Publication (PGP) ID Publication (PMID)           Publication (doi)  \\\n",
       "0                   PGP000001         25855707.0         10.1093/jnci/djv036   \n",
       "1                   PGP000001         25855707.0         10.1093/jnci/djv036   \n",
       "2                   PGP000001         25855707.0         10.1093/jnci/djv036   \n",
       "3                   PGP000002         30554720.0  10.1016/j.ajhg.2018.11.002   \n",
       "4                   PGP000002         30554720.0  10.1016/j.ajhg.2018.11.002   \n",
       "...                       ...                ...                         ...   \n",
       "3802                PGP000510         37268629.0  10.1038/s41467-023-38990-9   \n",
       "3803                PGP000510         37268629.0  10.1038/s41467-023-38990-9   \n",
       "3804                PGP000511         37429843.0  10.1038/s41467-023-39253-3   \n",
       "3805                PGP000512         37807951.0  10.1161/circgen.123.004259   \n",
       "3806                PGP000512         37807951.0  10.1161/circgen.123.004259   \n",
       "\n",
       "     Score and results match the original publication  \\\n",
       "0                                                True   \n",
       "1                                                True   \n",
       "2                                                True   \n",
       "3                                                True   \n",
       "4                                                True   \n",
       "...                                               ...   \n",
       "3802                                             True   \n",
       "3803                                             True   \n",
       "3804                                             True   \n",
       "3805                                             True   \n",
       "3806                                             True   \n",
       "\n",
       "      Ancestry Distribution (%) - Source of Variant Associations (GWAS)  \\\n",
       "0                                          European:100                   \n",
       "1                                          European:100                   \n",
       "2                                          European:100                   \n",
       "3                                          European:100                   \n",
       "4                                          European:100                   \n",
       "...                                                 ...                   \n",
       "3802  European:81|East Asian:11.3|African:5.2|Hispan...                   \n",
       "3803  European:79|East Asian:11|African:7.6|Hispanic...                   \n",
       "3804                                       European:100                   \n",
       "3805  European:94.1|Multi-ancestry (excluding Europe...                   \n",
       "3806  European:94.4|Multi-ancestry (excluding Europe...                   \n",
       "\n",
       "     Ancestry Distribution (%) - Score Development/Training  \\\n",
       "0                                                   NaN       \n",
       "1                                                   NaN       \n",
       "2                                                   NaN       \n",
       "3                                          European:100       \n",
       "4                                          European:100       \n",
       "...                                                 ...       \n",
       "3802  European:49.5|African:23.6|Hispanic or Latin A...       \n",
       "3803  European:62.6|African:16.8|Hispanic or Latin A...       \n",
       "3804                                                NaN       \n",
       "3805            Multi-ancestry (including European):100       \n",
       "3806            Multi-ancestry (including European):100       \n",
       "\n",
       "             Ancestry Distribution (%) - PGS Evaluation  \\\n",
       "0                                          European:100   \n",
       "1                                          European:100   \n",
       "2                                          European:100   \n",
       "3     European:76.1|Multi-ancestry (including Europe...   \n",
       "4     European:90|Not Reported:5|Multi-ancestry (exc...   \n",
       "...                                                 ...   \n",
       "3802                                                NaN   \n",
       "3803            Multi-ancestry (including European):100   \n",
       "3804                                       European:100   \n",
       "3805            Multi-ancestry (including European):100   \n",
       "3806            Multi-ancestry (including European):100   \n",
       "\n",
       "                                               FTP link Release Date  \\\n",
       "0     https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2019-10-14   \n",
       "1     https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2019-10-14   \n",
       "2     https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2019-10-14   \n",
       "3     https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2019-10-14   \n",
       "4     https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2019-10-14   \n",
       "...                                                 ...          ...   \n",
       "3802  https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2023-10-17   \n",
       "3803  https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2023-10-17   \n",
       "3804  https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2023-10-17   \n",
       "3805  https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2023-10-17   \n",
       "3806  https://ftp.ebi.ac.uk/pub/databases/spot/pgs/s...   2023-10-17   \n",
       "\n",
       "                                   License/Terms of Use  \n",
       "0     PGS obtained from the Catalog should be cited ...  \n",
       "1     PGS obtained from the Catalog should be cited ...  \n",
       "2     PGS obtained from the Catalog should be cited ...  \n",
       "3     PGS obtained from the Catalog should be cited ...  \n",
       "4     PGS obtained from the Catalog should be cited ...  \n",
       "...                                                 ...  \n",
       "3802  PGS obtained from the Catalog should be cited ...  \n",
       "3803  PGS obtained from the Catalog should be cited ...  \n",
       "3804  PGS obtained from the Catalog should be cited ...  \n",
       "3805  PGS obtained from the Catalog should be cited ...  \n",
       "3806  PGS obtained from the Catalog should be cited ...  \n",
       "\n",
       "[3807 rows x 21 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pgs_metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6e34c706-2364-4471-a0cb-7d2b851f517d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Trait (ontology term label)</th>\n",
       "      <th>Trait Identifier (ontology ID)</th>\n",
       "      <th>Trait Category</th>\n",
       "      <th>Number of Related PGS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abdominal Aortic Aneurysm</td>\n",
       "      <td>EFO_0004214</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abdominal pain</td>\n",
       "      <td>HP_0002027</td>\n",
       "      <td>Other trait</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abnormal circulating lipid concentration</td>\n",
       "      <td>HP_0003119</td>\n",
       "      <td>Other trait</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Abnormal EKG</td>\n",
       "      <td>HP_0003115</td>\n",
       "      <td>Other trait</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACPA-negative rheumatoid arthritis</td>\n",
       "      <td>EFO_0009460</td>\n",
       "      <td>Immune system disorder</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613</th>\n",
       "      <td>wellbeing measurement</td>\n",
       "      <td>EFO_0007869</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>white matter hyperintensity measurement</td>\n",
       "      <td>EFO_0005665</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>white matter volume measurement</td>\n",
       "      <td>EFO_0008320</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>whole body water mass</td>\n",
       "      <td>EFO_0009805</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>worry measurement</td>\n",
       "      <td>EFO_0009589</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>618 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Trait (ontology term label) Trait Identifier (ontology ID)  \\\n",
       "0                   Abdominal Aortic Aneurysm                    EFO_0004214   \n",
       "1                              Abdominal pain                     HP_0002027   \n",
       "2    Abnormal circulating lipid concentration                     HP_0003119   \n",
       "3                                Abnormal EKG                     HP_0003115   \n",
       "4          ACPA-negative rheumatoid arthritis                    EFO_0009460   \n",
       "..                                        ...                            ...   \n",
       "613                     wellbeing measurement                    EFO_0007869   \n",
       "614   white matter hyperintensity measurement                    EFO_0005665   \n",
       "615           white matter volume measurement                    EFO_0008320   \n",
       "616                     whole body water mass                    EFO_0009805   \n",
       "617                         worry measurement                    EFO_0009589   \n",
       "\n",
       "             Trait Category  Number of Related PGS  \n",
       "0    Cardiovascular disease                      3  \n",
       "1               Other trait                      3  \n",
       "2               Other trait                      2  \n",
       "3               Other trait                      1  \n",
       "4    Immune system disorder                      1  \n",
       "..                      ...                    ...  \n",
       "613       Other measurement                      4  \n",
       "614       Other measurement                      1  \n",
       "615       Other measurement                      4  \n",
       "616       Other measurement                      5  \n",
       "617       Other measurement                      4  \n",
       "\n",
       "[618 rows x 4 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trait_category_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "481794af-6236-4f55-b794-e685316acbca",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "PGS_EFO = dict(zip(pgs_metadata_df['Polygenic Score (PGS) ID'], pgs_metadata_df['Mapped Trait(s) (EFO ID)']))\n",
    "EFO_CAT = dict(zip(trait_category_df['Trait Identifier (ontology ID)'], trait_category_df['Trait Category']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cf3babad-9de2-4ea9-a0e0-210f30e7494d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#PGS_CAT translates all PGS_ids into disease categories\n",
    "PGS_CAT = dict()\n",
    "for pgs, efo in PGS_EFO.items():\n",
    "    PGS_CAT[pgs] = EFO_CAT[efo]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3c54aded-4fb8-4d35-ac46-d71de14379e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#PGS_CAT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2087057b-a35f-4957-9e60-0b26860501c2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PRS_name</th>\n",
       "      <th>intercept</th>\n",
       "      <th>PRS_q.value</th>\n",
       "      <th>X22009.0.1</th>\n",
       "      <th>X22009.0.2</th>\n",
       "      <th>X22009.0.3</th>\n",
       "      <th>X22009.0.4</th>\n",
       "      <th>X22009.0.5</th>\n",
       "      <th>X31.0.0</th>\n",
       "      <th>X21001.0.0</th>\n",
       "      <th>...</th>\n",
       "      <th>X54_11012</th>\n",
       "      <th>X54_11013</th>\n",
       "      <th>X54_11014</th>\n",
       "      <th>X54_11016</th>\n",
       "      <th>X54_11017</th>\n",
       "      <th>X54_11018</th>\n",
       "      <th>X54_11020</th>\n",
       "      <th>X54_11021</th>\n",
       "      <th>X54_11022</th>\n",
       "      <th>X54_11023</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_sumstats_ad_190513(29|Tlab_NonUKB)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Aragam_KG_PRS_241_Coronary_artery_disease_beta...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Aragam_KG_PRS_241_Coronary_artery_disease_join...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cad_190822_weights(168|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAD_GRS_300_JACC_2019_proxy(300|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2939</th>\n",
       "      <td>Tlab_TC_GLGC_PRS_74_TC_Nat_Genet_2013(73|Tlab_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2940</th>\n",
       "      <td>Tlab_TG_GLGC_PRS_40_TG_Nat_Genet_2013(40|Tlab_...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2941</th>\n",
       "      <td>Vanhoye_X_PRS_165_LDL_Translational_Resarch_20...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2942</th>\n",
       "      <td>Vujkovic_PRS_558_T2D_Nat_Genet(550|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2943</th>\n",
       "      <td>Ishan</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2944 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               PRS_name  intercept  \\\n",
       "0                 AD_sumstats_ad_190513(29|Tlab_NonUKB)        1.0   \n",
       "1     Aragam_KG_PRS_241_Coronary_artery_disease_beta...        1.0   \n",
       "2     Aragam_KG_PRS_241_Coronary_artery_disease_join...        NaN   \n",
       "3                   cad_190822_weights(168|Tlab_NonUKB)        NaN   \n",
       "4          CAD_GRS_300_JACC_2019_proxy(300|Tlab_NonUKB)        NaN   \n",
       "...                                                 ...        ...   \n",
       "2939  Tlab_TC_GLGC_PRS_74_TC_Nat_Genet_2013(73|Tlab_...        NaN   \n",
       "2940  Tlab_TG_GLGC_PRS_40_TG_Nat_Genet_2013(40|Tlab_...        1.0   \n",
       "2941  Vanhoye_X_PRS_165_LDL_Translational_Resarch_20...        1.0   \n",
       "2942    Vujkovic_PRS_558_T2D_Nat_Genet(550|Tlab_NonUKB)        NaN   \n",
       "2943                                              Ishan        1.0   \n",
       "\n",
       "      PRS_q.value  X22009.0.1  X22009.0.2  X22009.0.3  X22009.0.4  X22009.0.5  \\\n",
       "0        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "1        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "3        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "4        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "2939     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2940     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2941     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2942     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2943     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "\n",
       "      X31.0.0    X21001.0.0  ...  X54_11012  X54_11013  X54_11014  X54_11016  \\\n",
       "0     0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "1     0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "3     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "4     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "...       ...           ...  ...        ...        ...        ...        ...   \n",
       "2939  0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "2940  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2941  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2942  0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "2943  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "\n",
       "      X54_11017  X54_11018  X54_11020  X54_11021  X54_11022  X54_11023  \n",
       "0           1.0        1.0        1.0        1.0        1.0        1.0  \n",
       "1           1.0        1.0        1.0        1.0        1.0        1.0  \n",
       "2           NaN        NaN        NaN        NaN        NaN        NaN  \n",
       "3           NaN        NaN        NaN        NaN        NaN        NaN  \n",
       "4           NaN        NaN        NaN        NaN        NaN        NaN  \n",
       "...         ...        ...        ...        ...        ...        ...  \n",
       "2939        NaN        NaN        NaN        NaN        NaN        NaN  \n",
       "2940        1.0        1.0        1.0        1.0        1.0        1.0  \n",
       "2941        1.0        1.0        1.0        1.0        1.0        1.0  \n",
       "2942        NaN        NaN        NaN        NaN        NaN        NaN  \n",
       "2943        1.0        1.0        1.0        1.0        1.0        1.0  \n",
       "\n",
       "[2944 rows x 35 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#replace this by your pvalue table\n",
    "pvalue_table = pd.read_csv(\"results/hochberg_qvalues_21068_166900.csv\")\n",
    "pvalue_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ed4e5d8b-9a1c-4887-ba86-6e0a7a6e94f8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#convert each PGS id into their respective category names\n",
    "category_names = []\n",
    "for PRS_name in pvalue_table['PRS_name']:\n",
    "    PGS_id = PRS_name.split('_')[0]\n",
    "    if PGS_id in PGS_CAT.keys():\n",
    "        category_names.append(PGS_CAT[PGS_id])\n",
    "    elif PGS_id in ['cad', 'CAD', 'Aragam', 'Tlab', 'Nielsen', 'Vanhoye']:\n",
    "        category_names.append('Cardiovascular disease')\n",
    "    elif PGS_id in ['Vujkovic', 'Mahajan']:\n",
    "        category_names.append('Digestive system disorder')\n",
    "    elif PGS_id in ['AD', 'Lam']:\n",
    "        category_names.append('Neurological disorder')\n",
    "    else:\n",
    "        category_names.append('Other measurement')\n",
    "\n",
    "#add the category names to a column named \"Trait_category\"\n",
    "pvalue_table['Trait_category'] = category_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7ef2d2cd-bea7-4e8a-829f-cabb9e6a3d12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PRS_name</th>\n",
       "      <th>intercept</th>\n",
       "      <th>PRS_q.value</th>\n",
       "      <th>X22009.0.1</th>\n",
       "      <th>X22009.0.2</th>\n",
       "      <th>X22009.0.3</th>\n",
       "      <th>X22009.0.4</th>\n",
       "      <th>X22009.0.5</th>\n",
       "      <th>X31.0.0</th>\n",
       "      <th>X21001.0.0</th>\n",
       "      <th>...</th>\n",
       "      <th>X54_11013</th>\n",
       "      <th>X54_11014</th>\n",
       "      <th>X54_11016</th>\n",
       "      <th>X54_11017</th>\n",
       "      <th>X54_11018</th>\n",
       "      <th>X54_11020</th>\n",
       "      <th>X54_11021</th>\n",
       "      <th>X54_11022</th>\n",
       "      <th>X54_11023</th>\n",
       "      <th>Trait_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_sumstats_ad_190513(29|Tlab_NonUKB)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Neurological disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Aragam_KG_PRS_241_Coronary_artery_disease_beta...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Aragam_KG_PRS_241_Coronary_artery_disease_join...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cad_190822_weights(168|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAD_GRS_300_JACC_2019_proxy(300|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2939</th>\n",
       "      <td>Tlab_TC_GLGC_PRS_74_TC_Nat_Genet_2013(73|Tlab_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2940</th>\n",
       "      <td>Tlab_TG_GLGC_PRS_40_TG_Nat_Genet_2013(40|Tlab_...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2941</th>\n",
       "      <td>Vanhoye_X_PRS_165_LDL_Translational_Resarch_20...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cardiovascular disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2942</th>\n",
       "      <td>Vujkovic_PRS_558_T2D_Nat_Genet(550|Tlab_NonUKB)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Digestive system disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2943</th>\n",
       "      <td>Ishan</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999954</td>\n",
       "      <td>0.996844</td>\n",
       "      <td>0.112562</td>\n",
       "      <td>0.171108</td>\n",
       "      <td>0.983634</td>\n",
       "      <td>0.376903</td>\n",
       "      <td>0.01566</td>\n",
       "      <td>6.774339e-11</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Other measurement</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2944 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               PRS_name  intercept  \\\n",
       "0                 AD_sumstats_ad_190513(29|Tlab_NonUKB)        1.0   \n",
       "1     Aragam_KG_PRS_241_Coronary_artery_disease_beta...        1.0   \n",
       "2     Aragam_KG_PRS_241_Coronary_artery_disease_join...        NaN   \n",
       "3                   cad_190822_weights(168|Tlab_NonUKB)        NaN   \n",
       "4          CAD_GRS_300_JACC_2019_proxy(300|Tlab_NonUKB)        NaN   \n",
       "...                                                 ...        ...   \n",
       "2939  Tlab_TC_GLGC_PRS_74_TC_Nat_Genet_2013(73|Tlab_...        NaN   \n",
       "2940  Tlab_TG_GLGC_PRS_40_TG_Nat_Genet_2013(40|Tlab_...        1.0   \n",
       "2941  Vanhoye_X_PRS_165_LDL_Translational_Resarch_20...        1.0   \n",
       "2942    Vujkovic_PRS_558_T2D_Nat_Genet(550|Tlab_NonUKB)        NaN   \n",
       "2943                                              Ishan        1.0   \n",
       "\n",
       "      PRS_q.value  X22009.0.1  X22009.0.2  X22009.0.3  X22009.0.4  X22009.0.5  \\\n",
       "0        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "1        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "3        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "4        0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "...           ...         ...         ...         ...         ...         ...   \n",
       "2939     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2940     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2941     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2942     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "2943     0.999954    0.996844    0.112562    0.171108    0.983634    0.376903   \n",
       "\n",
       "      X31.0.0    X21001.0.0  ...  X54_11013  X54_11014  X54_11016  X54_11017  \\\n",
       "0     0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "1     0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "3     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "4     0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "...       ...           ...  ...        ...        ...        ...        ...   \n",
       "2939  0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "2940  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2941  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "2942  0.01566  6.774339e-11  ...        NaN        NaN        NaN        NaN   \n",
       "2943  0.01566  6.774339e-11  ...        1.0        1.0        1.0        1.0   \n",
       "\n",
       "      X54_11018  X54_11020  X54_11021  X54_11022  X54_11023  \\\n",
       "0           1.0        1.0        1.0        1.0        1.0   \n",
       "1           1.0        1.0        1.0        1.0        1.0   \n",
       "2           NaN        NaN        NaN        NaN        NaN   \n",
       "3           NaN        NaN        NaN        NaN        NaN   \n",
       "4           NaN        NaN        NaN        NaN        NaN   \n",
       "...         ...        ...        ...        ...        ...   \n",
       "2939        NaN        NaN        NaN        NaN        NaN   \n",
       "2940        1.0        1.0        1.0        1.0        1.0   \n",
       "2941        1.0        1.0        1.0        1.0        1.0   \n",
       "2942        NaN        NaN        NaN        NaN        NaN   \n",
       "2943        1.0        1.0        1.0        1.0        1.0   \n",
       "\n",
       "                 Trait_category  \n",
       "0         Neurological disorder  \n",
       "1        Cardiovascular disease  \n",
       "2        Cardiovascular disease  \n",
       "3        Cardiovascular disease  \n",
       "4        Cardiovascular disease  \n",
       "...                         ...  \n",
       "2939     Cardiovascular disease  \n",
       "2940     Cardiovascular disease  \n",
       "2941     Cardiovascular disease  \n",
       "2942  Digestive system disorder  \n",
       "2943          Other measurement  \n",
       "\n",
       "[2944 rows x 36 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pvalue_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "afbc5018-47aa-40b5-add2-fe175d7c745b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#save the pvalue table with the new column added\n",
    "#change the file path and name below according to your own pvalue result file\n",
    "pvalue_table.to_csv(\"results/hochberg_qvalues_21068_166900_trait_cat.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06734df3-47d3-4553-b786-f123fad807cc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "UFRC Python-3.10",
   "language": "python",
   "name": "python3-3.10-ufrc"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
